Labs and Factory

plaidrabbit opened the pull request.

Commit:
Adding factory and lab functions

Commit:
Adding in a Commune Class, with the LabManager hanging off of it.

Commit:
cleanup

Commit:
more cleanup

Commit:
more cleanup

Commit:
Merge remote-tracking branch 'origin/Development' into Development

Commit:
Fix for loading the Commune.

Commit:
Making the lab IDs not hard coded.
Removing a lot of the reservation code, to start removing it.  This way we're not modifiying the server objects.
Fixing where it wouldn't always pull energy when trying to fill mandatory energy items.

Commit:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

Commit:
More of getting rid of code that modifies the screeps server object's storage.

CarsonBurke reviewed (changes requested):
Please reverse all changes to links.ts, fastFiller.ts, and creepFunctions.ts

Commit:
Merge remote-tracking branch 'origin/Development' into Development

Commit:
Merge remote-tracking branch 'origin/Development' into Development

Commit:
Bunch of fixes.  Getting it to spawn another hauler a little earlier
some fixes around early game.
Making the temp default lab action G
Fixed a glitch in one of the factory conditions
Checking for some error/edge cases
adding stack traces into the error message
fixing a compiler error
doing a workaround for the freeCapacityAfterTick to replace reservations

Commit:
Couple small fixes that were in my other branch

Commit:
Merge remote-tracking branch 'remotes/origin/Development' into Development
